src/sentry/api/endpoints/organization_details.py

```diff
@@ -516,6 +522,35 @@ def put(self, request, organization):
                     data=changed_data,
                 )
 
+                # Temporarily writing org-level apdex changes to ProjectTransactionThreshold
+                # for orgs who don't have the feature enabled so that when this
+                # feature is GA'ed it captures the orgs current apdex threshold.
+                if "apdexThreshold" in changed_data and not features.has(
+                    "organizations:project-transaction-threshold", organization
+                ):
+                    match = re.match(r".*to (\d+)$", changed_data["apdexThreshold"])
+                    if match:
+                        apdex_threshold = int(match.group(1))
+                        ProjectTransactionThreshold.objects.filter(
+                            organization_id=organization.id
+                        ).delete()
+
+                        project_ids = Project.objects.filter(
+                            organization_id=organization.id
+                        ).values_list("id", flat=True)
+
+                        ProjectTransactionThreshold.objects.bulk_create(
+                            [
+                                ProjectTransactionThreshold(
+                                    project_id=project_id,
+                                    organization_id=organization.id,
+                                    threshold=int(apdex_threshold),
+                                    metric=TransactionMetric.DURATION.value,
+                                )
+                                for project_id in project_ids
+                            ]
+                        )
```

Comment:
Would be good to have these in a transaction so that you don't end up with no rows should the delete succeed and of of the inserts hit an integrity error because of concurrent writes.

Reply:
In your comment, you selected only these lines of code:
```
                        project_ids = Project.objects.filter(
                            organization_id=organization.id
                        ).values_list("id", flat=True)

                        ProjectTransactionThreshold.objects.bulk_create(
                            [
                                ProjectTransactionThreshold(
                                    project_id=project_id,
                                    organization_id=organization.id,
                                    threshold=int(apdex_threshold),
                                    metric=TransactionMetric.DURATION.value,
                                )
                                for project_id in project_ids
                            ]
                        )
```
But the delete also needs to be in the transaction so it gets rolled back in case of an Integrity error right?

Reply:
Yes. Sorry for the bad range selection.